src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -0,0 +1,302 @@
+
+#include <cstdlib>
+#include <iostream>
+#include <sqlite3.h>
+#include <tuple>
+#include <vector>
+
+#include "src/fairness/weighted_tree/test/load_accounting_db.hpp"
+
+using namespace Flux::accounting;
+
+std::shared_ptr<weighted_tree_node_t> node = nullptr;
+std::shared_ptr<weighted_tree_node_t> user_node = nullptr;
+std::shared_ptr<weighted_tree_node_t> root_ptr = nullptr;
+
+std::shared_ptr<weighted_tree_node_t>
+get_sub_banks (sqlite3 *DB,
+               const std::string &bank_name,
+               std::shared_ptr<weighted_tree_node_t> parent_bank,
+               sqlite3_stmt *b_select_shares_stmt,
+               sqlite3_stmt *b_select_sub_banks_stmt,
+               sqlite3_stmt *b_select_associations_stmt) {
+    int rc = 0;
+    int leaf_bank_usage = 0;
+
+    // bind parameter to prepare SQL statement
+    rc = sqlite3_bind_text (b_select_shares_stmt,
+                            1,
+                            bank_name.c_str (),
+                            -1,
+                            NULL);
+    if (rc != SQLITE_OK) {
+        fprintf (stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+    // execute SQL statement and store result
+    rc = sqlite3_step (b_select_shares_stmt);
+    if (rc == SQLITE_ERROR) {
+        fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+
+    std::string bank_shares = reinterpret_cast<char const *>(
+        sqlite3_column_text (b_select_shares_stmt, 0));
+
+    // initialize a weighted tree node
+    node = std::make_shared<weighted_tree_node_t> (parent_bank,
+                                                   bank_name,
+                                                   false,
+                                                   std::stoll (bank_shares),
+                                                   0);
+
+    // if there is no parent bank, then the root pointer points to the root bank
+    if (!parent_bank) {
+        root_ptr = node;
+    } else {
+        parent_bank->add_child (node);
+    }
+
+    rc = sqlite3_bind_text (b_select_sub_banks_stmt,
+                            1,
+                            bank_name.c_str (),
+                            -1,
+                            NULL);
+    if (rc != SQLITE_OK) {
+        fprintf (stderr, "Failed to fetch data: %s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+    rc = sqlite3_step (b_select_sub_banks_stmt);
+    if (rc == SQLITE_ERROR) {
+        fprintf (stderr, "%s\n", sqlite3_errmsg (DB));
+        sqlite3_close (DB);
+
+        return nullptr;
+    }
+
+    // vector of strings to hold sub banks
+    std::vector<std::string> banks;
+
+    // we've reached a bank with no sub banks, so print
+    // out all associations under this sub bank
+    if (rc != SQLITE_ROW) {
```

Comment:
This is a huge conditional statement. Would it be possible to refactor the logic into some functions to improve readability?